### PR TITLE
[NodeChanger] Create MethodNameChanger

### LINKED ISSUE #201: [Proposal] Rector\NodeChanger
As [suggested](https://github.com/rectorphp/rector/pull/198#discussion_r157882742) in #198 by @TomasVotruba, we can see the following pattern in `Rector`s:
```php
if($condition) {
    $methodCallNode->name = new Identifier($newMethodName);
} else {
    $methodCallNode->name = new Identifier($OtherNewMethodName);
}
```
This could be change to a service called `Rector\NodeChanger\MethodNameChanger` :man_technologist:

### PR DESCRIPTION
**Pull request by carusogabriel**
This resolves #201, creating a `NodeChanger\MethodNameChanger` to delegate the function or rename a Node method name. Implemented it in `SpecificMethodCountRector` for tests, but if @TomasVotruba agreed, let's constantly use in `Rector`s.

### CODE REVIEW

File: src/NodeChanger/MethodNameChanger.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+use PhpParser\Node;
+use PhpParser\Node\Identifier;
+
+final class MethodNameChanger
+{
+    /**
+     * @param string[] $renameMethodMap
+     */
+    public function renameNode(Node $node, array $renameMethodMap): ?Node
```
Comment (TomasVotruba) [inline src/NodeChanger/MethodNameChanger.php]: I think `Node` type can be `MethodCall`

File: src/Rector/Contrib/PHPUnit/SpecificMethodCountRector.php [hunk 2]
```diff
@@ -87,9 +93,7 @@ public function isCandidate(Node $node): bool
      */
     public function refactor(Node $methodCallNode): ?Node
     {
-        $oldMethodName = $methodCallNode->name->toString();
-
-        $methodCallNode->name = new Identifier($this->renameMethodsMap[$oldMethodName]);
+        $this->methodNameChanger->renameNode($methodCallNode, $this->renameMethodsMap);
```
Comment (TomasVotruba) [inline src/Rector/Contrib/PHPUnit/SpecificMethodCountRector.php]: Nice!

**Comment by TomasVotruba** [comment]
Thank you for nice and clear PR! 

Could you add to other places where is the rename done manually?

Mainly to these `if/else`

**Comment by carusogabriel** [comment]
@TomasVotruba Sure, gonna simplify `if/else`s as much as I can :man_technologist:

**Comment by TomasVotruba** [comment]
Looks great!

Thanks for contribution

**Pull request merged**